components/brave_wallet_ui/common/hardware/types.ts

```diff
@@ -35,60 +33,178 @@ export type SignHardwareOperationResult = HardwareOperationResult & {
   payload?: EthereumSignedTx | FilSignedLotusMessage | Buffer | string
 }
 
-export enum LedgerDerivationPaths {
-  LedgerLive = 'ledger-live',
-  Legacy = 'legacy',
-  Deprecated = 'deprecated'
+export const DerivationSchemes = {
+  EthLedgerLive: 'EthLedgerLive',
+  EthLedgerLegacy: 'EthLedgerLegacy',
+  EthLedgerDeprecated: 'EthLedgerDeprecated',
+  SolLedgerDefault: 'SolLedgerDefault',
+  SolLedgerLedgerLive: 'SolLedgerLedgerLive',
+  SolLedgerBip44Root: 'SolLedgerBip44Root',
+  FilLedgerMainnet: 'FilLedgerMainnet',
+  FilLedgerTestnet: 'FilLedgerTestnet',
+  BtcLedgerMainnet: 'BtcLedgerMainnet',
+  BtcLedgerTestnet: 'BtcLedgerTestnet',
+
+  EthTrezorDefault: 'EthTrezorDefault'
+} as const
+
+export type DerivationScheme =
+  (typeof DerivationSchemes)[keyof typeof DerivationSchemes]
+
+export interface HardwareImportScheme {
+  derivationScheme: DerivationScheme
+  coin: BraveWallet.CoinType
+  keyringId: BraveWallet.KeyringId
+  vendor: BraveWallet.HardwareVendor
+  name: string
+  /** Some schemes are valid only for a single network for corresponding
+   * coin. */
+  fixedNetwork?: string
+  /** For some schemes there is no indexing in template and only one account is
+   * supported. */
+  singleAccount?: boolean
+  pathTemplate: (index: number | 'x') => string
+}
+
+export const EthLedgerLiveHardwareImportScheme: HardwareImportScheme = {
+  derivationScheme: DerivationSchemes.EthLedgerLive,
+  coin: BraveWallet.CoinType.ETH,
+  keyringId: BraveWallet.KeyringId.kDefault,
+  vendor: BraveWallet.HardwareVendor.kLedger,
+  name: 'Ledger Live',
+  pathTemplate: (index) => `m/44'/60'/${index}'/0/0`
+}
+
+export const EthLedgerLegacyHardwareImportScheme: HardwareImportScheme = {
+  derivationScheme: DerivationSchemes.EthLedgerLegacy,
+  coin: BraveWallet.CoinType.ETH,
+  keyringId: BraveWallet.KeyringId.kDefault,
+  vendor: BraveWallet.HardwareVendor.kLedger,
+  name: 'Legacy (MEW/MyCrypto)',
+  pathTemplate: (index) => `m/44'/60'/0'/${index}`
+}
+
+export const EthLedgerDeprecatedHardwareImportScheme: HardwareImportScheme = {
+  derivationScheme: DerivationSchemes.EthLedgerDeprecated,
+  coin: BraveWallet.CoinType.ETH,
+  keyringId: BraveWallet.KeyringId.kDefault,
+  vendor: BraveWallet.HardwareVendor.kLedger,
+  name: 'Deprecated (Not recommended)',
```

Comment:
Should we localize this name?

Reply:
It didn't have l10 before, I've just moved such strings to a new place. Added an issue to fix https://github.com/brave/brave-browser/issues/40148